arches/app/models/models.py

```diff
@@ -1246,6 +1233,22 @@ def get_initial_resource_instance_lifecycle_state(self, *args, **kwargs):
         User, on_delete=models.SET_NULL, blank=True, null=True
     )
 
+    class Meta:
+        managed = True
+        db_table = "resource_instances"
+        permissions = (("no_access_to_resourceinstance", "No Access"),)
+
+    def __init__(self, *args, **kwargs):
+        super(ResourceInstance, self).__init__(*args, **kwargs)
+        if not self.resourceinstanceid:
+            self.resourceinstanceid = uuid.uuid4()
```

Comment:
I moved this from the bottom of the class because I sort of expect to see it up before the other methods, but if others feel different, happy to shuffle it back.